src/dmqproto/node/neo/request/Consume.d

```diff
@@ -1298,36 +1296,70 @@ public abstract scope class ConsumeProtocol_v3
             );
         }
 
+        /***********************************************************************
+
+            Sends the current batch of records to the client and waits for a
+            `Continue` or  `Stopped` feedback. Sending can be interrupted by a
+            `Stop` signal. Clears the batch of records if it has been sent. If
+            `false` is returned and `(*this.record_batch).length == 0` then the
+            batch of records has not been sent because it was interrupted by a
+            `Stop` signal.
+
+            Returns:
+                - `true` if the current batch of records has been sent to the
+                   client and the client responded with `Continue`,
+                - `false` if the current batch of records has been sent to the
+                   client and the client responded with `Stop`,
+                - `false`  `false` if interrupted by a `Stop` signal, meaning
+                  the `Stop` has been received while waiting to send.
+
+        ***********************************************************************/
+
+        private bool sendBatchAndWaitForFeedback ( )
+        {
+            auto event = this.outer.request_event_dispatcher.nextEvent(
+                this.fiber,
+                Send(&this.fillInBatchPayload), Signal(SignalCode.Stop)
```

Comment:
Hmm... there's something wrong here. The point of having the two separate fibers is to avoid having to care about handling every possible type of event each time you suspend.

I mean, of course the main thing is that it works and isn't too complicated, but I get the feeling there's a way to make this code simpler. Maybe we can have a look at this together later. It's always possible we've found a case where having two fibers really doesn't help. It'd be interesting to investigate.